src/elections/urls.py

```diff
@@ -0,0 +1,160 @@
+import base64
+import logging
+import os
+import re
+import urllib.parse
+from datetime import datetime
+
+import requests  # TODO: make this async
+import xmltodict
+from crud import ElectionParameters
+from fastapi import APIRouter, BackgroundTasks, FastAPI, HTTPException, Request, status
+from fastapi.exceptions import RequestValidationError
+from tables import election_types
+
+import auth
+import auth.crud
+import database
+import elections
+from constants import root_ip_address
+from permission import types
+
+_logger = logging.getLogger(__name__)
+
+router = APIRouter(
+    prefix="/elections",
+    tags=["elections"],
+)
+
+def _slugify(
+        text: str
+) -> str:
+    """
+    Creates a unique slug based on text passed in. Assumes non-unicode text.
+    """
+    return re.sub(r"[\W_]+", "-", text)
+
+async def _validate_user(
+        db_session: database.DBSession,
+        session_id: str
+) -> dict:
+    computing_id = await auth.crud.get_computing_id(db_session, session_id)
+    # Assuming now user is validated
+    result = await types.ElectionOfficer.has_permission(db_session, computing_id)
+    return result
+
+@router.get(
+    "/create_election",
+    description="Creates an election and places it in the database",
+)
+async def create_election(
+    request: Request,
+    db_session: database.DBSession,
+    name: str,
+    election_type: str,
+    start_datetime: datetime | None = None,
+    end_datetime: datetime | None = None,
+    survey_link: str | None = None
+):
+    """
+    aaa
+    """
+    session_id = request.cookies.get("session_id", None)
+    user_auth = await _validate_user(db_session, session_id)
+    if user_auth is False:
+        # let's workshop how we actually wanna handle this
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="You do not have permission to access this resource",
+            headers={"WWW-Authenticate": "Basic"},
+        )
+
+    # Default start time should be now unless specified otherwise
+    if start_datetime is None:
+        start_datetime = datetime.now()
+
+    if election_type not in election_types:
+        raise RequestValidationError()
+
+    params = ElectionParameters(
+        _slugify(name),
```

Comment:
since `slug` is used as a unique identifier, what if we have a duplicate slug here ("My election" vs "my election", etc...)? We should have a check to make sure that slug does not currently exist, so we don't accidentally overwrite a past election.